algorithms/gr-bfsrh/csharp/BreadthFirst.cs

```diff
@@ -0,0 +1,96 @@
+using System.Collections.Generic;
+
+namespace Algorithms
+{
+    //Based on https://www.geeksforgeeks.org/shortest-distance-two-cells-matrix-grid/
+    //Binary-first search
+    public class BreadthFirst
+    {
+        private const int BLOCKED = 1;
+        private const int START = 2;
+        private const int END = 3;
+
+        public static int MinDistance(int[][] grid)
+        {
+            Node source = new Node(0, 0, 0);
+
+            var n = grid.Length;
+            var m = grid[0].Length;
+
+            // To keep track of visited QItems. Marking 
+            // blocked cells as visited. 
+            bool[][] visited = TrackVisited(grid, source, n, m);
+
+            // applying BFS on matrix cells starting from source 
+            Queue<Node> q = new Queue<Node>();
+            q.Enqueue(source);
+            visited[source.X][source.Y] = true;
+            while (q.Count > 0)
+            {
+                var p = q.Dequeue();
+
+                // Destination found; 
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6319078)